polar-core/src/vm.rs

```diff
@@ -3934,4 +4023,35 @@ mod tests {
             QueryEvent::Done { result: true }
         ]);
     }
+
+    #[test]
+    fn test_log_level_should_print_for_level() {
+        let trace = LogLevel::Trace;
+        let info = LogLevel::Info;
+        let debug = LogLevel::Debug;
+
+        // TRACE
+        assert!(trace.should_print_on_level(trace), "trace prints trace");
+        assert!(trace.should_print_on_level(debug), "trace prints debug");
+        assert!(trace.should_print_on_level(info), "trace prints info");
+
+        // DEBUG
+        assert!(
+            !debug.should_print_on_level(trace),
+            "debug does not print trace"
+        );
+        assert!(debug.should_print_on_level(debug), "debug prints debug");
+        assert!(debug.should_print_on_level(info), "debug prints info");
+
+        // INFO
+        assert!(
+            !info.should_print_on_level(trace),
+            "info does not print trace"
+        );
+        assert!(
+            !info.should_print_on_level(debug),
+            "info does not print debug"
+        );
+        assert!(info.should_print_on_level(info), "info prints info");
```

Comment:
truly the smallest of nits: the "on assertion error" messages aren't really making it much easier to understand each individual case, and I think having them all lined up like 
```suggestion
        assert!(!info.should_print_on_level(trace));
        assert!(!info.should_print_on_level(debug));
        assert!(info.should_print_on_level(info));
```
would be easier to quickly visually grep